internal/vm/compiler.go

```diff
@@ -17,7 +17,7 @@ import (
 // Compile compiles a program from the input into a virtual machine or a list
 // of compile errors.  It takes the program's name and the metric store as
 // additional arguments to build the virtual machine.
-func Compile(name string, input io.Reader, emitAst bool, emitAstTypes bool, syslogUseCurrentYear bool, loc *time.Location) (*VM, error) {
+func Compile(name string, input io.Reader, emitAst bool, emitAstTypes bool, syslogUseCurrentYear bool, loc *time.Location, maxRegexLength int, maxRecursionDepth int) (*VM, error) {
```

Comment:
This now looks like a prime candidate for Options pattern, I'll add that in a followup change.  Can you leave a TODO here?